src/llmcompressor/modeling/gpt_oss.py

```diff
@@ -0,0 +1,249 @@
+        )
+
+    @torch.no_grad()
+    def copy_from_fused_weights(
+        self,
+        legacy_gate_up_W: torch.Tensor,  # [E, H, 2D]
+        legacy_gate_up_b: torch.Tensor,  # [E, 2D]
+        legacy_down_W: torch.Tensor,     # [E, D, H]
+        legacy_down_b: torch.Tensor,     # [E, H]
+    ) -> None:
+        """
+        De-interleave fused gate_up weights/bias and copy into separate gate/up experts.
+        """
+        E, H, twoD = legacy_gate_up_W.shape
+        assert E == self.num_experts
+        D = twoD // 2
+        assert D == self.expert_dim
+
+        for i in range(E):
+            Wi = legacy_gate_up_W[i]   # [H, 2D]
+            bi = legacy_gate_up_b[i]   # [2D]
+
+            Wg = Wi[:, 0::2].contiguous()  # [H, D]
+            Wu = Wi[:, 1::2].contiguous()  # [H, D]
+            bg = bi[0::2].contiguous()     # [D]
+            bu = bi[1::2].contiguous()     # [D]
+
+            expert = self.experts[i]
+            expert.gate_proj.weight.copy_(Wg.t())
+            expert.gate_proj.bias.copy_(bg)
+            expert.up_proj.weight.copy_(Wu.t())
+            expert.up_proj.bias.copy_(bu)
+
+            expert.down_proj.weight.copy_(legacy_down_W[i].t())
+            expert.down_proj.bias.copy_(legacy_down_b[i])
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,         # [B, T, H]
+        router_indices: Optional[torch.Tensor] = None,   # [B, T, top_k] or [tokens, top_k]
+        routing_weights: Optional[torch.Tensor] = None,  # [B, T, E] or [tokens, E]
+    ) -> torch.Tensor:
+        """
+        Implements the MoE computation using the router outputs.
+
+        This is compatible with the GPT-OSS MoE call pattern:
+            experts(hidden_states, router_indices, routing_weights)
+        """
+        assert routing_weights is not None and router_indices is not None, "router inputs required"
+
+        # Normalize shapes to [tokens, H], [tokens, top_k], [tokens, E]
+        if hidden_states.dim() == 3:
+            B, T, H = hidden_states.shape
+            x = hidden_states.reshape(-1, H)
+        else:
+            # Already flattened
+            B, T = 1, hidden_states.shape[0]
+            H = hidden_states.shape[-1]
+            x = hidden_states
+
+        if router_indices.dim() == 3:
+            router_indices = router_indices.reshape(-1, router_indices.shape[-1])
+        if routing_weights.dim() == 3:
+            routing_weights = routing_weights.reshape(-1, routing_weights.shape[-1])
+
+        num_experts_plus_dummy = routing_weights.shape[1]
+        out = torch.zeros_like(x)
+
+        # GPT-OSS router uses an extra "no expert" bucket at index E
+        with torch.no_grad():
+            expert_mask = torch.nn.functional.one_hot(
+                router_indices, num_classes=num_experts_plus_dummy
+            ).permute(2, 1, 0)
+            expert_hit = torch.greater(expert_mask.sum(dim=(-1, -2)), 0).nonzero()
+
+        for idx in expert_hit:
```

Comment:
In the future, we'd like to expand this to support the `calibrate_all_experts` option. However for this example, the forward function is not used anyways, so this is fine.